### DISCUSSION TITLE
Clarification on Room Assistant GPIO

### DISCUSSION THREAD

**Discussion by Menethoran** [mKeRix/room-assistant#399] [Q&A]
Hey guys... so... in an attempt to figure out whats wrong with my build, i realized theres some not friendly wording on the way to set up PIR sensors in regards to Room Assistant... so...

When told to write the configuration and when getting to the "PIN" category, are we looking to put in the physical PIN number for the attached PIR output... or are we looking to put the GPIO number there (For example, a raspberry Pi Pin #16 is GPIO23)... So, do we use "16" or 23" in that position?
Second, how do we setup a pi to configure that pin as an output (or is that built into RA/NodeJ?)

Sorry, the literature for physical sensors and RA  isnt great.... and makes this a bit difficult (then theres the logs... which are sparse... makes it difficult to track down where the errors are happening)

Thanks in advance.

**Comment by mKeRix** [comment]
If I remember correctly it should be `23` in that case, because the underlying library is using the `/sys/class/gpio` interface. I just opened up the documentation again and agree that it could be improved. If you want feel free to open a PR and add your learnings afterwards, otherwise I'll try to add some information based on these questions when I find the time (and remember). :)

**Reply by mKeRix** [reply]
Oh and for outputs: those are always switches, so configuring a pin under `binarySensors` makes it an input while configuring it under `switches` makes it an output.

**Comment by Menethoran** [comment]
Outside of the scope of the post ...
Did Raspbian Buster change the way these things operate? I set a motion
sensor up like this for a DAKboard a year or so ago and didn't have any
trouble .. I'm fighting this thing left and right trying to figure out what
it wants (I alsmot feel like I have bad sensors or something)...

Because, correct me if I'm wrong, I want to use 'PIN' because I want the
input from the PIR sensor... Right? Like, I do not want "Switch" because
that would be for sending a return command to say... Turn on an LED or
something?
(Sorry, ripping my hair out over this .. and I'm already bald ;) )

**Reply by mKeRix** [reply]
I’m not aware of any changes related to GPIO on Buster, but to be fair the last time that I interacted with that was while I was testing the integration nearly a year ago. 😅

Exactly, you want it to be an input, so a binary sensor.
